components/camel-sip/src/main/java/org/apache/camel/component/sip/listener/SipSubscriptionListener.java

```diff
@@ -34,48 +35,106 @@
+    /**
+     * The potential dialog between this retrieving SIP point and the sender
+     */
     private Dialog subscriberDialog;
+
+    /**
+     * The dialog for when the sender made one
+     */
     private Dialog forkedDialog;
 
-    public SipSubscriptionListener(SipSubscriber sipSubscriber) {
-        this.setSipSubscriber(sipSubscriber);
+    /**
+     * Create a SipListener listening for subscription events
+     * @param sipConsumer the camel consumer creating this listener
+     */
+    public SipSubscriptionListener(SipConsumer sipConsumer) {
+        this.setSipConsumer(sipConsumer);
     }
 
+    /**
+     * Sends the message body from a SIP request event along the camel route.
+     * @param response the message body from the SIP request
+     * @throws CamelException when dispatching the message goes wrong
+     */
     private void dispatchExchange(Object response) throws CamelException {
         LOG.debug("Consumer Dispatching the received notification along the route");
-        Exchange exchange = sipSubscriber.getEndpoint().createExchange(ExchangePattern.InOnly);
+        Exchange exchange = sipConsumer.getEndpoint().createExchange(ExchangePattern.InOnly);
         exchange.getIn().setBody(response);
-        try {
-            sipSubscriber.getProcessor().process(exchange);
-        } catch (Exception e) {
+
+        //try to send the message along
+        try
+        {
+            sipConsumer.getProcessor().process(exchange);
+        }
+        catch (Exception e)
+        {
             throw new CamelException("Error in consumer while dispatching exchange", e);
         }
     }
-    
+
+    /**
+     * process incoming SIP requests. Only processes NOTIFY and MESSAGE requests
+     * @param requestReceivedEvent the request event
+     */
     public void processRequest(RequestEvent requestReceivedEvent) {
         Request request = requestReceivedEvent.getRequest();
         ServerTransaction serverTransactionId = requestReceivedEvent
                 .getServerTransaction();
         String viaBranch = ((ViaHeader)(request.getHeaders(ViaHeader.NAME).next())).getParameter("branch");
-        LOG.debug("Request: {}", request.getMethod()); 
+        String fromBranch = ((FromHeader)(request.getHeader(FromHeader.NAME))).toString();
+
+        //log the retrieved request
+        LOG.debug("Request: {}", request.getMethod());
         LOG.debug("Server Transaction Id: {}", serverTransactionId);
         LOG.debug("Received From Branch: {}", viaBranch);
+        LOG.debug("Received From: {}", fromBranch);
 
-        if (request.getMethod().equals(Request.NOTIFY)) {
+        //process NOTIFY and MESSAGE requests only
+        if (Request.NOTIFY.equals(request.getMethod()))
+        {
             processNotify(requestReceivedEvent, serverTransactionId);
-        } 
+        }
+        else if(Request.MESSAGE.equals(request.getMethod()))
+        {
+            processMessage(requestReceivedEvent, serverTransactionId);
```

Comment:
@nikvaessen as it is SIP "SIMPLE", maybe this method should be called ```processSimpleMessage``` to make it more obvious what it is